Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -607,6 +607,26 @@ void HTMLScriptElement::unmark_as_parser_inserted(Badge<DOM::Range>)
     m_parser_document = nullptr;
 }
 
+[[nodiscard]] String HTMLScriptElement::src() const
+{
+    return get_attribute(HTML::AttributeNames::src).value_or(String {});
+}
+
+void HTMLScriptElement::set_src(String const& src)
+{
+    auto old_src = get_attribute(HTML::AttributeNames::src);
+
+    MUST(set_attribute(HTML::AttributeNames::src, src));
+
+    // https://html.spec.whatwg.org/multipage/scripting.html#script-processing-model
+    // When a script element el that is not parser-inserted experiences one of the events listed in the following list, the user agent must immediately prepare the script element el:
+    // - [...]
+    // - The script element is connected and has a src attribute set where previously the element had no such attribute.
+    if (!is_parser_inserted() && is_connected() && !old_src.has_value()) {
+        prepare_script();
+    }
```

Comment:
I'm not sure this will work when calling `element.setAttribute("src", <value>)`  from JS. I think moving this logic into `attribute_changed()` should allow this to work when calling `setAttribute()` and when setting the property directly.

Reply:
Moved to `attribute_changed`, although is there maybe a better way to check what was the old value of `src` (instead of just saving a flag in the `HTMLScriptElement`)?

Reply:
Perhaps `attribute_changed()` could take both the old and new value as parameters?

Reply:
Eventually yeah, but this would require changing this everywhere. IMO this is out of scope of this PR. (Also are there any other places which require this?)